kano_profile/tracker/tracking_uuids.py

```diff
@@ -0,0 +1,179 @@
+            to use as an expiry timestamp. Only used when creating a new uuid.
+
+    Returns:
+        dict: A uuid object. See :func:`._new_tracking_uuid`.
+    """
+
+    uuid = str(uuid5(uuid1(), key))
+
+    tracking_uuid = _read_tracking_uuid(key)
+
+    if not tracking_uuid or _is_uuid_expired(tracking_uuid):
+        tracking_uuid = _new_tracking_uuid(uuid, expires)
+        _add_tracking_uuid(key, tracking_uuid)
+
+    return tracking_uuid
+
+
+def _new_tracking_uuid(uuid, expires):
+    """Create a new uuid object with the given parameters.
+
+    Args:
+        uuid (str): A randomised hash code.
+        expires (int): See :func:`.get_tracking_uuid`.
+
+    Returns:
+        dict: A uuid object.
+    """
+
+    timestamp = time.time()
+
+    return {
+        'uuid': uuid,
+        'timestamp': timestamp,
+        'expires': timestamp + expires
+    }
+
+
+def _is_uuid_expired(tracking_uuid):
+    """Check whether a given uuid object is expired.
+
+    Args:
+        tracking_uuid (dict): A uuid object.
+
+    Returns:
+        bool: Whether the uuid is expired.
+    """
+
+    expired = time.time() > tracking_uuid.get('expires', 0)
+    logger.debug('tracking_uuid {} expired = {}'.format(tracking_uuid, expired))
+
+    return expired
+
+
+def _read_tracking_uuid(key):
+    """Retrieve the key associated uuid from storage.
+
+    Args:
+        key (str): The namespace associated with a uuid, e.g. 'kano-tracker-ctl'.
+
+    Returns:
+        dict: A uuid object if one is found or an empty dict otherwise.
+    """
+
+    uuids_file, data = _open_uuids()
+    uuids_file.close()
+
+    return data.get(key, dict())
+
+
+def _add_tracking_uuid(key, tracking_uuid):
+    """Store a new uuid object.
+
+    Args:
+        key (str): The namespace associated with a uuid, e.g. 'kano-tracker-ctl'.
+    """
+
+    uuids_file, data = _open_uuids()
+
+    with uuids_file:
+        data[key] = tracking_uuid
+        uuids_file.seek(0)
+        uuids_file.truncate(0)
+        uuids_file.write(json.dumps(data))
+
+    if 'SUDO_USER' in os.environ:
+        chown_path(TRACKER_UUIDS_PATH)
+
+
+def remove_tracking_uuid(key):
+    """Remove the uuid object associated with the given key.
+
+    Args:
+        key (str): The namespace associated with a uuid, e.g. 'kano-tracker-ctl'.
+    """
+
+    uuids_file, data = _open_uuids()
+
+    with uuids_file:
+        if key in data:
+            data.pop(key)
+            uuids_file.seek(0)
+            uuids_file.truncate(0)
+            uuids_file.write(json.dumps(data))
+
+    if 'SUDO_USER' in os.environ:
+        chown_path(TRACKER_UUIDS_PATH)
+
+
+def _open_uuids():
+    """Helper function to open the uuids file and load the JSON inside.
+
+    Returns:
+        file, dict: The opened uuids file and its JSON data.
+    """
+
+    newly_created = False
+    data = dict()
+
+    if not os.path.exists(TRACKER_UUIDS_PATH):
+        touch(TRACKER_UUIDS_PATH)
+        newly_created = True
+
+    try:
```

Comment:
is this try-except block properly indented?

Reply:
Yup, it is a try-except-else block which apparently is valid Python :)